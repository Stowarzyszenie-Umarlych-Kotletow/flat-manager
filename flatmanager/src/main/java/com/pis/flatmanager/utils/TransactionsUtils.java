package com.pis.flatmanager.utils;

import com.pis.flatmanager.model.transactions.MinMaxValue;

import java.math.BigDecimal;
import java.util.Collection;
import java.util.Map;

public class TransactionsUtils {

    private TransactionsUtils() {

    }

    public static <T> MinMaxValue<T> getMinMax(Collection<Map.Entry<T, BigDecimal>> decimals) {
        var min = BigDecimal.ZERO;
        var max = BigDecimal.ZERO;
        T minObj = null;
        T maxObj = null;
        for (var kv : decimals) {
            var key = kv.getKey();
            var value = kv.getValue();
            if (value.compareTo(max) > 0) {
                max = value;
                maxObj = key;
            }
            if (value.compareTo(min) < 0) {
                min = value;
                minObj = key;
            }
        }
        return new MinMaxValue<>(minObj, min, maxObj, max);
    }
}
