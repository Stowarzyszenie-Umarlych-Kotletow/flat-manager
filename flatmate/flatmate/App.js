import * as React from 'react';
import {View, Text, Button} from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {LoginScreen} from './views/Accounts/LoginScreen';
import {RegisterScreen} from './views/Accounts/RegisterScreen';
import {HomeScreen} from './views/Accounts/HomeScreen';
import {ManageScreen} from './views/Accounts/ManageScreen';
import {ChangePasswordScreen} from './views/Accounts/ChangePasswordScreen';


const Stack = createNativeStackNavigator();

function App() {
    return (
        <NavigationContainer>
            <Stack.Navigator initialRouteName="HomeScreen">
                <Stack.Screen name="HomeScreen" component={HomeScreen} options={{
                    title: 'Flatmate'
                }} />
                <Stack.Screen name="RegisterScreen" component={RegisterScreen} options={{
                    title: 'Register'
                }} />
                <Stack.Screen name="LoginScreen" component={LoginScreen} options={{
                    title: 'Login'
                }} />
                <Stack.Screen name="ManageScreen" component={ManageScreen} options={{
                    title: 'Manage account'
                }} />
                <Stack.Screen name="ChangePasswordScreen" component={ChangePasswordScreen} options={{
                    title: 'Change Password'
                }} />
            </Stack.Navigator>
        </NavigationContainer>
    );
}


export default App;