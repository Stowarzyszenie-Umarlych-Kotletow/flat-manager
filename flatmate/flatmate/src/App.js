import * as React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {LoggedNavigator} from './LoggedNavigator';
import {createStackNavigator} from '@react-navigation/stack';
import 'react-native-gesture-handler';
import {NotLoggedNavigator} from './NotLoggedNavigator'

const Stack = createStackNavigator(); // NativeStackNavigator is maybe better



function App() {
    const [loggedUser, setUser] = React.useState(null)
    return (
        <NavigationContainer>
            <Stack.Navigator
                screenOptions={{
                    headerShown: false,
                }}>
                {loggedUser == null ? (
                    <Stack.Screen name="NotLoggedNavigator" options={{
                        title: 'NotLoggedNavigator',
                    }}>
                        {
                            props=><NotLoggedNavigator {...props} {...{setUser}}/>
                        }
                    </Stack.Screen>
                ) : (
                    <Stack.Screen name="LoggedNavigator" options={{
                        title: 'LoggedNavigator'
                    }}>
                        {
                            props=><LoggedNavigator {...props} {...{setUser}}/>
                        }
                    </Stack.Screen>
                ) }
            </Stack.Navigator>
        </NavigationContainer>
    );
}


export default App;