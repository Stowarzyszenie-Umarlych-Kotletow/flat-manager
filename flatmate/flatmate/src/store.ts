import { combineReducers, configureStore, applyMiddleware } from '@reduxjs/toolkit'
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import storage from '@react-native-async-storage/async-storage';
import auth from "./features/auth"
import flat from './features/flat';
import { api } from './features/api/api';
import { setupListeners } from '@reduxjs/toolkit/dist/query';

const rootReducer = combineReducers({
  auth: auth.reducer,
  flat: flat.reducer,
  [api.reducerPath]: api.reducer
});
const persistConfig = {
  key: 'root',
  storage
};
const persistedReducer = persistReducer(persistConfig, rootReducer);


const store = configureStore({
  reducer: persistedReducer,
  devTools: true,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }).concat(api.middleware)
});

setupListeners(store.dispatch);

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>() // Export a hook that can be reused to resolve types
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
export default store;