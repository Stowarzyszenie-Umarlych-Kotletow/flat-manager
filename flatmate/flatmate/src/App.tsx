import * as React from 'react';
import 'react-native-gesture-handler';
import { Provider } from 'react-redux';
import store, { persistor, useAppDispatch, useAppSelector} from "./store";
import { PersistGate } from 'redux-persist/integration/react';
import RootContainer from "./RootContainer";
import { ModalPortal } from 'react-native-modals';


function App() {
    return (
        <Provider store={store}>
            <PersistGate persistor={persistor}>
                <RootContainer />
                <ModalPortal />
            </PersistGate>
        </Provider>
    );
}


export default App;