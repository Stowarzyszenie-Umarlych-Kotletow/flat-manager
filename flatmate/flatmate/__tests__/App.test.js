import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mockStore } from "./utils/test-utils";
import store from "../src/store";
import { ModalPortal } from 'react-native-modals';
import App from '../src/App'

test('App Screen renders correctly', () => {
  const tree = renderer.create(<Provider store={mockStore()}><App /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});