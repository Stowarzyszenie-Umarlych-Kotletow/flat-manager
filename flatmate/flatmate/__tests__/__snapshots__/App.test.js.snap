// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Screen renders correctly 1`] = `null`;
